### PR TITLE
fix #864;  allow password to be entered twice to verify

### LINKED ISSUE #864: Enhancement: new wallet password confirmation
Simplewallet should ask the user to retype the password when creating a new one.

Suggested by xxxgoodgirls on bitcointalk

### PR DESCRIPTION
**Pull request by guzzijones**
Password prompt will show up once if the wallet exists.
Password prompt will show up twice if the wallet does not exist.
Cleaned up an extra constructor that only took a password.  It was not being used.
Squashed commits into one commit.

Let me know.

### CODE REVIEW

**Comment by guzzijones** [comment]
I think "EC2 Default User" ended up in there from squashing the commits.  Oh well.

File: src/simplewallet/password_container.cpp [hunk 1]
```diff
@@ -132,6 +131,41 @@ namespace tools
...
+bool password_container::read_from_tty_double_check(const char *message) {
+    std::string pass1;
+    std::string pass2;
+    bool match=false;
+    do{
+        if (message)
+            std::cout << message <<": ";
+        if (!password_container::read_from_tty(pass1))
+            return false;
+        if (m_wallet_file_name.empty()){
```
Comment (guzzijones) [inline src/simplewallet/password_container.cpp]: The user has entered a wallet file name so do not double check the password.

File: src/simplewallet/password_container.cpp [hunk 2]
```diff
@@ -48,24 +48,25 @@ namespace tools
...
   password_container::password_container()
     : m_empty(true)
   {
-  }
 
-  password_container::password_container(std::string&& password)
-    : m_empty(false)
-    , m_password(std::move(password))
+  }
+  password_container::password_container(const std::string m_wallet_file)
```
Comment (moneromooo-monero) [inline src/simplewallet/password_container.cpp]: forgot a reference I think

Comment (moneromooo-monero) [inline src/simplewallet/password_container.cpp]: Also, I don't like that a wallet file name is passed to that. It's got no business with that. Apparently that's just to check if it's empty, so it should be a bool that's passed, not a string to check emptiness.

**Comment by guzzijones** [comment]
redoing

**Pull request closed without merging**